cmake/gtest_utils.cmake

```diff
@@ -0,0 +1,75 @@
+enable_testing()
+
+function(k2_add_gtest)
+  cmake_parse_arguments(
+      ARGS                                 # prefix of output variables
+      "NO_MAIN;NO_GMOCK;EXCLUDE_FROM_ALL"  # list of names of the boolean arguments (only defined ones will be true)
+      ""                                   # list of names of mono-valued arguments
+      "SOURCES;DEPENDS"                    # list of names of multi-valued arguments (output variables are lists)
```

Comment:
I think `gtest`, `gmock`, `gtest_main` and `gmock_main` all can be passed via
`DEPENDS` and there is no need to use separate options for `NO_MAIN`, `NO_GMOCK`,
which can simplify this function significantly.

Reply:
Done.